pkg/cmd/cli/secrets/add_secret_to_obj.go

```diff
@@ -149,33 +180,37 @@ func (o AddSecretOptions) Validate() error {
 }
 
 func (o AddSecretOptions) AddSecrets() error {
-	r := resource.NewBuilder(o.Mapper, o.Typer, o.ClientMapper).
-		NamespaceParam(o.Namespace).
-		ResourceTypeOrNameArgs(false, o.TargetName).
-		SingleResourceType().
-		Do()
-	if r.Err() != nil {
-		return r.Err()
-	}
-	obj, err := r.Object()
-	if err != nil {
-		return err
-	}
-
-	switch t := obj.(type) {
-	case *api.ServiceAccount:
-		switch o.Type {
-		case PullType:
-			_, err := o.AddSecretsToSAPullSecrets(t)
-			return err
-		case MountType:
-			_, err := o.AddSecretsToSAMountableSecrets(t)
+	for _, secretType := range o.TypeList {
```

Comment:
adding to both pull and mount secrets should be done in a single API update, not two